datadog/resource_datadog_synthetics_test_test.go

```diff
@@ -877,6 +933,164 @@ resource "datadog_synthetics_test" "ssl" {
 }`, uniq)
 }
 
+func createSyntheticsTCPTestStep(accProvider *schema.Provider, clock clockwork.FakeClock, t *testing.T) resource.TestStep {
+	testName := uniqueEntityName(clock, t)
+	return resource.TestStep{
+		Config: createSyntheticsTCPTestConfig(testName),
+		Check: resource.ComposeTestCheckFunc(
+			testSyntheticsTestExists(accProvider),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "type", "api"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "subtype", "tcp"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "request.host", "agent-intake.logs.datadoghq.com"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "request.port", "443"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "assertions.#", "1"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "assertions.0.type", "responseTime"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "assertions.0.operator", "lessThan"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "assertions.0.target", "2000"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "locations.#", "1"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "locations.0", "aws:eu-central-1"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "options.tick_every", "60"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "name", testName),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "message", "Notify @datadog.user"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "tags.#", "2"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "tags.0", "foo:bar"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "tags.1", "baz"),
+			resource.TestCheckResourceAttr(
+				"datadog_synthetics_test.tcp", "status", "paused"),
+			resource.TestCheckResourceAttrSet(
+				"datadog_synthetics_test.tcp", "monitor_id"),
+		),
+	}
+}
+
+func createSyntheticsTCPTestConfig(uniq string) string {
+	return fmt.Sprintf(`
+resource "datadog_synthetics_test" "tcp" {
+	type = "api"
+	subtype = "tcp"
+
+	request = {
+		host = "agent-intake.logs.datadoghq.com"
+		port = 443
+	}
+
+	assertions = [
```

Comment:
Can you use the new `assertion` instead? This one is deprecated. You can remove the stateignore in your test too  I think.